asset-transfer-ledger-queries/chaincode-go/asset_transfer_ledger_chaincode.go

```diff
@@ -388,30 +388,32 @@ func getQueryResultForQueryStringWithPagination(ctx contractapi.TransactionConte
 
 // GetAssetHistory returns the chain of custody for an asset since issuance.
 func (t *SimpleChaincode) GetAssetHistory(ctx contractapi.TransactionContextInterface, assetID string) ([]QueryResult, error) {
-
 	resultsIterator, err := ctx.GetStub().GetHistoryForKey(assetID)
 	if err != nil {
 		return nil, err
 	}
 	defer resultsIterator.Close()
 
-	records := []QueryResult{}
-
+	var records []QueryResult
 	for resultsIterator.HasNext() {
 		response, err := resultsIterator.Next()
 		if err != nil {
 			return nil, err
 		}
 
-		asset := new(asset)
-		err = json.Unmarshal(response.Value, asset)
+		var asset *asset
+		err = json.Unmarshal(response.Value, &asset)
 		if err != nil {
 			return nil, err
 		}
 
+		timestamp, err := ptypes.Timestamp(response.Timestamp)
```

Comment:
a comment on  why to use ptypes.Timestamp will be useful

Reply:
I'm not sure what a comment would solve here. `response.Timestamp` is a protobuf, so we use the protobuf type rather than build it ourselves which is unconventional. I feel a comment would have been more justified for the other implementation where the `unix.Time` was used, as this has the potential to lose timezone related information.